be/src/fs/fs_starlet.cpp

```diff
@@ -119,40 +154,53 @@ class StarletFileSystem : public FileSystem {
 
     Type type() const override { return STARLET; }
 
+    StatusOr<Configuration> get_shard_config(int64_t tablet_id) {
+        Configuration conf;
+        ASSIGN_OR_RETURN(auto shardinfo, g_worker->get_shard_info(tablet_id));
+        conf[kS3AccessKeyId] = shardinfo.properties[kS3AccessKeyId];
+        conf[kS3AccessKeySecret] = shardinfo.properties[kS3AccessKeySecret];
+        conf[kS3OverrideEndpoint] = shardinfo.properties[kS3OverrideEndpoint];
+        conf[kS3Bucket] = shardinfo.properties[kS3Bucket];
+        return conf;
+    }
+
     StatusOr<std::unique_ptr<RandomAccessFile>> new_random_access_file(const std::string& path) override {
         return new_random_access_file(RandomAccessFileOptions(), path);
     }
 
     StatusOr<std::unique_ptr<RandomAccessFile>> new_random_access_file(const RandomAccessFileOptions& opts,
                                                                        const std::string& path) override {
-        auto format_str = parse_starlet_path(path);
-        DCHECK(g_starlet != nullptr);
-        staros::starlet::ObjectStorePtr object_store = g_starlet->get_store(format_str);
-        if (object_store == nullptr) {
-            return Status::InternalError(fmt::format("Failed to get store from starlet path {}", path));
+        ASSIGN_OR_RETURN(auto pair, parse_starlet_path(path));
+        ASSIGN_OR_RETURN(auto conf, get_shard_config(pair.second))
+
+        auto fs_st = FileSystemFactory::new_filesystem("s3://", conf);
+        if (!fs_st.ok()) {
+            return to_status(fs_st.status());
         }
-        auto st = object_store->new_object(format_str);
-        if (!st.ok()) {
-            return to_status(st.status());
+
+        auto file_st = (*fs_st)->open(pair.first, ReadOptions());
+
+        if (!file_st.ok()) {
+            return to_status(file_st.status());
         }
-        auto object = *st;
-        auto istream = std::make_shared<StarletInputStream>(object->get_input_stream());
+        auto istream = std::make_shared<StarletInputStream>(std::move(*file_st));
         return std::make_unique<RandomAccessFile>(std::move(istream), path);
     }
 
     StatusOr<std::unique_ptr<SequentialFile>> new_sequential_file(const std::string& path) override {
-        auto format_str = parse_starlet_path(path);
-        DCHECK(g_starlet != nullptr);
-        staros::starlet::ObjectStorePtr object_store = g_starlet->get_store(format_str);
-        if (object_store == nullptr) {
-            return Status::InternalError(fmt::format("Failed to get store from starlet path {}", path));
+        ASSIGN_OR_RETURN(auto pair, parse_starlet_path(path));
+        ASSIGN_OR_RETURN(auto conf, get_shard_config(pair.second))
+
+        auto fs_st = FileSystemFactory::new_filesystem("s3://", conf);
```

Comment:
expect to get filesystem according to the shard_info, `new_filesystem(conf[kSysRoot], conf)`